coreneuron/io/nrn_setup.cpp

```diff
@@ -1186,952 +867,22 @@ void delete_trajectory_requests(NrnThread& nt) {
-        nt._vecplay[i] = new VecPlayContinuous(ml->data + ix, yvec, tvec, nullptr, nt.id);
-    }
-    if (!direct) {
-        // store current checkpoint state to continue reading mapping
-        F.record_checkpoint();
-
-        // If not at end of file, then this must be a checkpoint and restore tqueue.
-        if (!F.eof()) {
-            checkpoint_restore_tqueue(nt, F);
-        }
-    }
-
-    // NetReceiveBuffering
-    for (auto& net_buf_receive : corenrn.get_net_buf_receive()) {
-        int type = net_buf_receive.second;
-        // Does this thread have this type.
-        Memb_list* ml = nt._ml_list[type];
-        if (ml) {  // needs a NetReceiveBuffer
-            NetReceiveBuffer_t* nrb =
-                (NetReceiveBuffer_t*)ecalloc_align(1, sizeof(NetReceiveBuffer_t));
-            ml->_net_receive_buffer = nrb;
-            nrb->_pnt_offset = pnt_offset[type];
-
-            // begin with a size of 5% of the number of instances
-            nrb->_size = ml->nodecount;
-            // or at least 8
-            nrb->_size = std::max(8, nrb->_size);
-            // but not more than nodecount
-            nrb->_size = std::min(ml->nodecount, nrb->_size);
-
-            nrb->_pnt_index = (int*)ecalloc_align(nrb->_size, sizeof(int));
-            nrb->_displ = (int*)ecalloc_align(nrb->_size + 1, sizeof(int));
-            nrb->_nrb_index = (int*)ecalloc_align(nrb->_size, sizeof(int));
-            nrb->_weight_index = (int*)ecalloc_align(nrb->_size, sizeof(int));
-            nrb->_nrb_t = (double*)ecalloc_align(nrb->_size, sizeof(double));
-            nrb->_nrb_flag = (double*)ecalloc_align(nrb->_size, sizeof(double));
-        }
-    }
-
-    // NetSendBuffering
-    for (int type : corenrn.get_net_buf_send_type()) {
-        // Does this thread have this type.
-        Memb_list* ml = nt._ml_list[type];
-        if (ml) {  // needs a NetSendBuffer
-            NetSendBuffer_t* nsb = (NetSendBuffer_t*)ecalloc_align(1, sizeof(NetSendBuffer_t));
-            ml->_net_send_buffer = nsb;
-
-            // begin with a size equal to twice number of instances
-            // at present there is no provision for dynamically increasing this.
-            nsb->_size = ml->nodecount * 2;
-            nsb->_cnt = 0;
-
-            nsb->_sendtype = (int*)ecalloc_align(nsb->_size, sizeof(int));
-            nsb->_vdata_index = (int*)ecalloc_align(nsb->_size, sizeof(int));
-            nsb->_pnt_index = (int*)ecalloc_align(nsb->_size, sizeof(int));
-            nsb->_weight_index = (int*)ecalloc_align(nsb->_size, sizeof(int));
-            // when == 1, NetReceiveBuffer_t is newly allocated (i.e. we need to free previous copy
-            // and recopy new data
-            nsb->reallocated = 1;
-            nsb->_nsb_t = (double*)ecalloc_align(nsb->_size, sizeof(double));
-            nsb->_nsb_flag = (double*)ecalloc_align(nsb->_size, sizeof(double));
-        }
-    }
+void read_phase1(FileHandler& F, int imult, NrnThread& nt) {
```

Comment:
Those wrappers are usefull to capture global variable declared in `nrn_setup.cpp` (here `corenrn_embedded`), because the resolution of phase is done in `nrn_setup.hpp` where it does not exist